### ISSUE TITLE
Some Generic Method Signatures Result in Syntactically Incorrect Generated Code

### ISSUE THREAD

**Issue by shabino** [immutables/immutables#686]
When using Immutables 2.5.6 (did not test with previous releases), the following two classes result in syntactically incorrect generated code.

```
@Immutable
public abstract class AbstractBar<T> {
}

@Immutable
public abstract class AbstractFoo {
   public abstract Bar<?> bar();
}
```

The generated code looks like this:

```
public final class ImmutableFoo extends AbstractFoo {
  private final <any> bar;

  private ImmutableFoo(<any> bar) {
    this.bar = bar;
  }
  <rest of generated code omitted>
}
```
The generated code for ImmutableBar was correct. Replacing `Bar<?>` with `List<?>` resulted in correctly-generated code.

**Comment by vojtechhabarta** [comment]
I have similar problem, just instead of `Bar<?>` I want `Bar<String>`, `Bar<Integer>` etc.

My use case is similar to https://immutables.github.io/immutable.html#wrapper-types but instead of creating derived types like `PersonName` I want to use directly `Wrapper<String>`. In my case purpose of the `Wrapper` is not to improve type safety but to distinguish following cases in some request object:
- request property change to some value (using `Wrapper.of(someValue)`)
- request property change to `null` (using `Wrapper.of(null)`)
- request no property change (using `null`)

Example:
``` java
@Value.Immutable
public interface UserUpdateRequest {
    String name();
    @Nullable Wrapper<Integer> age();
    @Nullable Wrapper<String> address();
}
```

I tried versions 2.5.5 and 2.5.6.

**Comment by elucash** [comment]
this was a long-standing issue with generic arguments when used in attributes together with not-yet-generated types, which is hopefully fixed (#548) and available in 2.6 milestones (currently 2.6.0-alpha3, 2.6.0 is to be released soon)
